# Confidence Intervals

In [1]:
# Libraries
import pandas as pd
import numpy as np
import scipy.stats as st
from scipy.stats import norm
import math

## Challenge 1
We want to estimate the average size of the men of a country with a confidence level of 80%. Assuming that the standard deviation of the sizes in the population is 4, get the confidence interval with a sample of men selected randomly, whose heights are:

````
heights = [167, 167, 168, 168, 168, 169, 171, 172, 173, 175, 175, 175, 177, 182, 195]
````

**Hint**: function `stats.norm.interval` from `scipy` can help you get through this exercise. 

In [2]:
# your code here
heights = [167, 167, 168, 168, 168, 169, 171, 172, 173, 175, 175, 175, 177, 182, 195]

std_dev_population = 4

sem = std_dev_population / np.sqrt(len(heights))

interval = st.norm.interval(alpha=0.8, loc=np.mean(heights), scale=sem)

print(f'There is a 80% chance that the confidence interval of {interval} contains the true population mean heights.')

There is a 80% chance that the confidence interval of (172.14308590115726, 174.79024743217607) contains the true population mean heights.


C:\Users\Yuliya Lavrenyuk\AppData\Local\Temp\ipykernel_6992\1066444004.py:8: DeprecationWarning: Use of keyword argument 'alpha' for method 'interval' is deprecated and wil be removed in SciPy 1.11.0. Use first positional argument or keyword argument 'confidence' instead.
  interval = st.norm.interval(alpha=0.8, loc=np.mean(heights), scale=sem)


In [10]:
st.t.interval(alpha=0.8, df=len(heights)-1, loc=np.mean(heights), scale=sem)

C:\Users\Yuliya Lavrenyuk\AppData\Local\Temp\ipykernel_26276\3274047935.py:1: DeprecationWarning: Use of keyword argument 'alpha' for method 'interval' is deprecated and wil be removed in SciPy 1.11.0. Use first positional argument or keyword argument 'confidence' instead.
  st.t.interval(alpha=0.8, df=len(heights)-1, loc=np.mean(heights), scale=sem)


(172.07752526922508, 174.85580806410826)

## Challenge 2 
In a sample of 105 shops selected randomly from an area, we note that 27 of them have had losses in this month. Get an interval for the proportion of businesses in the area with losses to a confidence level of 80% and a confidence level of 90%.

**Hint**: function `stats.norm.interval` from `scipy` can help you get through this exercise. 

In [11]:
# your code here

n = 105
x = 27

sample_proportion = x / n

In [13]:
SE = math.sqrt((sample_proportion * (1 - sample_proportion)) / n)

# Confidence intervals using stats.norm.interval
confidence_level_80 = 0.80
confidence_level_90 = 0.90

CI_80 = st.norm.interval(confidence_level_80, loc=sample_proportion, scale=SE)
CI_90 = st.norm.interval(confidence_level_90, loc=sample_proportion, scale=SE)

# Results
print(f"80% Confidence Interval: {CI_80}")
print(f"90% Confidence Interval: {CI_90}")

80% Confidence Interval: (0.20248138545542083, 0.3118043288302934)
90% Confidence Interval: (0.1869856177645281, 0.3273000965211861)


We can be 80% confident that the true proportion of businesses with losses is between approximately 20.2% and 31.2%, and 90% confident that it is between approximately 18.6% and 32.8%.

## Bonus Challenge
The error level or sampling error for the first challenge is given by the following expression:
$$Error = z_{\frac{\alpha}{2}}\frac{\sigma}{\sqrt n}$$
Where z represents the value for N(0,1)


Suppose that with the previous data of challenge 1, and with a confidence level of
99% (that is, almost certainly) we want to estimate the average population size, so that the error level committed is not greater than half a centimeter.

#### 1.- Determine what size the selected sample of men should be.

#### 2.- For the second challenge, we have the following error:
$$ Error = z_{\frac{\alpha}{2}}\sqrt{\frac{p\times q}{n}} $$
#### Determine the sample size required to not exceed an error of 1% with a confidence of 80%.

In [ ]:
# your code here

## Bonus Challenge

Let's consider the following problem:

Build a confidence interval of 94% for the real difference between the durations of two brands of spotlights, if a sample of 40 spotlights taken randomly from the first mark gave an average duration of 418 hours, and a sample of 50 bulbs of another brand gave a duration average of 402 hours. The standard deviations of the two
populations are 26 hours and 22 hours, respectively.

Sometimes, we will be interested in the difference of two different groups of random variables. We can also build a confidence interval for that! We have some different cases regarding the variance but for this specific case (the variance are different and known), we have that:

$$\overline{X} - \overline{Y}  \sim N(\mu_{X} - \mu_{Y} , \sqrt{\frac{\sigma_{X}^2}{n_X}+\frac{\sigma_{Y}^2}{n_Y}})$$

Solve the problem with this information.

In [ ]:
# your code here